packages/app-desktop/gui/ConfigScreen/controls/plugins/useOnInstallHandler.test.ts

```diff
@@ -0,0 +1,97 @@
+import useOnInstallHandler from './useOnInstallHandler';
+import { renderHook } from '@testing-library/react-hooks';
+
+import PluginService, { defaultPluginSetting, PluginSetting } from '@joplin/lib/services/plugins/PluginService';
+import { ItemEvent } from './PluginBox';
+import produce from 'immer';
+
+jest.mock('@joplin/lib/services/plugins/PluginService');
+jest.mock('immer');
+
+const pluginServiceinstance = {
+	updatePluginFromRepo: jest.fn(),
+	installPluginFromRepo: jest.fn(),
+};
+
+const pluginId = 'test.plugin';
+const setInstallingPluginIds = jest.fn();
+const repoApi = jest.fn();
+const onPluginSettingsChange = jest.fn();
+const itemEvent = ({
+	item: { manifest: { id: pluginId } },
+} as ItemEvent);
+const callHook = (isUpdate: boolean, pluginEnabled = true) => () => useOnInstallHandler(
+	setInstallingPluginIds,
+	{
+		[pluginId]: {
+			enabled: pluginEnabled,
+			deleted: false,
+			hasBeenUpdated: false,
+		},
+	},
+	repoApi,
+	onPluginSettingsChange,
+	isUpdate
+);
+
+describe('useOnInstallHandler', () => {
+
+	beforeAll(() => {
+		(PluginService.instance as jest.Mock).mockReturnValue(pluginServiceinstance);
+		(defaultPluginSetting as jest.Mock).mockImplementation(
+			jest.requireActual('@joplin/lib/services/plugins/PluginService').defaultPluginSetting
+		);
+	});
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	test('should set updatingPluginIds while updating', async () => {
```

Comment:
In general please could you give a more higher level description? For example "should report that the plugin is being updated". This applies to all the tests - as much as possible there shouldn't be variable names such as "updatingPluginIds" or "hasBeenUpdated" in the description, and instead you should give a plain English description of what's being tested. Something that even someone non-technical can read and it would make sense.

If it's not easy to give such description, it might mean that the test is too low level. We don't aim for 100% test coverage and favour integration tests (although integration tests are not possible in this case).